### ISSUE TITLE
Move Scala-Arc to it's own subdirectory

### ISSUE THREAD

**Issue by segeljakt** [cda-group/arc#26]
The current directory structure places Scala-Arc at the root and MLIR-Arc in a subdirectory. What do you @frej think about placing Scala-Arc in its own subdirectory also? It's very minor, but I think it makes things more consistent.

For example:
```
arc-frontend/
  project/
  src/
  .scalafmt.conf
  build.sbt
  scalastyle-config.xml
arc-mlir/
docs/
.gitignore
.gitmodules
.travis.yml
README.md
```

**Comment by frej** [comment]
Fine by me, no objections at all.

**Comment by segeljakt** [comment]
For my next PR, I want to refactor the structure of the project. CI will be covered in a later commit once we have the mega computer.

Currently, the directory structure looks like this (some details not included):
```
arc/
  src/
    main/
    test/
  build.sbt
  arc-mlir/
    src/
      include/
        arc/
          arc-dialect.h
          arc-ops.td
      arc-dialect.cpp
      arc-opts.cpp
      arc-opts.td
      arc-to-mlir.in
      main.cpp
    arc-mlir-build
  README.md
```
I propose we change it into:
```
arc/
  arc-frontend/
    build.sbt
    src/
      main/
      test/
  arc-mlir/
    src/
      include/
        arc/
          Types.h
          Dialect.h
          Ops.td
      lib/
        Dialect.cpp
        Opts.cpp
        Opts.td
      to-mlir.in
      Main.cpp
    arc-mlir-build
  README.md
```
What do you think @frej?

**Comment by frej** [comment]
I'm all for it, introducing a lib directory for the dialect is as upstream does it.

**Comment by segeljakt** [comment]
I'm wondering @frej, maybe it is also worth dividing the `arc-mlir` test directory into:

```
test/
  arc/ # Contains tests for converting from Arc into MLIR
  ops/ # Contains one file of tests per operation
  types/ # Contains one file of tests per type
  opts/ # Contains one file of tests per optimisation
```

**Comment by frej** [comment]
Go for it!

**Comment by segeljakt** [comment]
This issue can be closed with #57.
